server/src/main/java/org/elasticsearch/index/fielddata/IndexFieldDataService.java

```diff
@@ -106,14 +108,27 @@ public synchronized void clearField(final String fieldName) {
         ExceptionsHelper.maybeThrowRuntimeAndSuppress(exceptions);
     }
 
+    /**
+     * Returns fielddata for the provided field type. Same as {@link #getForField(MappedFieldType, String, Supplier)} but does not take
+     * the index name and the search lookup supplier as arguments. Does not support runtime fields.
+     */
     public <IFD extends IndexFieldData<?>> IFD getForField(MappedFieldType fieldType) {
-        return getForField(fieldType, index().getName());
+        assert fieldType.isRuntimeField() == false : "runtime fields are not supported";
```

Comment:
I am not sure this is needed, index sort will check that the field has doc values so it will fail at the Lucene level. I am not saying we shouldn't check at the ES level but this `assert`  is confusing IMO. Can we remove this function entirely and replace it with the one below ?

Reply:
correct, this is to have a proper error message. But what I am after here is to protect from this method being a second `getForField` which though can't support runtime fields (we have no search lookup!). It is so easy to use this one instead of the other one. Maybe renaming both would also help. 

>Can we remove this function entirely and replace it with the one below?

you mean having a single getForField method I assume? That would be great but I am not sure where to get search lookup for this usecase that does not need it and it's only used for index warmers and index sorting.

Reply:
I removed the getForField(MappedFieldType). Callers now call the other getForField and provide their own search lookup supplier and index name. That removed the need for the isRuntimeField flag too. index sorting already throws a decent error when it is configured with a runtime field (doc values not found for field etc.). I am happier with this as it simplifies things.